kratos/elements/embedded_nodal_variable_calculation_element_simplex.h

```diff
@@ -0,0 +1,432 @@
+        const NodesArrayType& ThisNodes)
+        : Element(NewId, ThisNodes)
+    {}
+
+    /// Constructor using a geometry object.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     */
+    EmbeddedNodalVariableCalculationElementSimplex(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry)
+        : Element(NewId, pGeometry)
+    {}
+
+    /// Constuctor using geometry and properties.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     * @param pProperties Pointer to the element's properties
+     */
+    EmbeddedNodalVariableCalculationElementSimplex(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry,
+        PropertiesType::Pointer pProperties)
+        : Element(NewId, pGeometry, pProperties)
+    {}
+
+    /// Destructor.
+    ~EmbeddedNodalVariableCalculationElementSimplex() override
+    {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Create a new element of this type
+    /**
+     * Returns a pointer to a new EmbeddedNodalVariableCalculationElementSimplex element, created using given input
+     * @param NewId the ID of the new element
+     * @param ThisNodes the nodes of the new element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_shared<EmbeddedNodalVariableCalculationElementSimplex<TVarType>>(NewId, GetGeometry().Create(ThisNodes), pProperties);
+    }
+
+    void Initialize() override
+    {
+    }
+
+    /// Calculate the element's local contribution to the system for the current step.
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(
+        MatrixType &rLeftHandSideMatrix,
+        ProcessInfo &rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(
+        VectorType &rRightHandSideVector,
+        ProcessInfo &rCurrentProcessInfo) override;
+
+    /// Checks the input and that all required Kratos variables have been registered.
+    /**
+     * This function provides the place to perform checks on the completeness of the input.
+     * It is designed to be called only once (or anyway, not often) typically at the beginning
+     * of the calculations, so to verify that nothing is missing from the input
+     * or that no common error is found.
+     * @param rCurrentProcessInfo The ProcessInfo of the ModelPart that contains this element.
+     * @return 0 if no errors were found.
+     */
+    int Check(const ProcessInfo &rCurrentProcessInfo) override
```

Comment:
please remove if it does nothing

Reply:
I forgot to implement it. Now it does some checks.